src/main/java/core/basesyntax/impl/FruitshopServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.database.Operation;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitshopService;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.OperationStrategy;
+import java.util.List;
+import java.util.Map;
+
+public class FruitshopServiceImpl implements FruitshopService, OperationStrategy {
+    private final Map<Operation, OperationHandler> operationHandlerMap;
+
+    public FruitshopServiceImpl(Map<Operation, OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public void processData(List<FruitTransaction> fruitTransactionList,
+                            Map<Operation, OperationHandler> operationHandlerMap) {
+        for (FruitTransaction value : fruitTransactionList) {
+            getHandler(value).apply(value.getFruit(), value.getQuantity());
+        }
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction fruitTransaction) {

```

Comment:
delete this method

Reply:
Why? Without this method a can't get handler type and apply strategy to each operation type. I watched video from theory, and mentor implemented task the same way..